containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceListener.java

```diff
@@ -70,7 +71,61 @@ public void stop() {
      * @return The id of the namespace or null if the namespace does not exist
      */
     public String getNamespaceId(String namespaceName) {
-        return namespaces.get(namespaceName);
+        String namespaceId = namespaces.get(namespaceName);
+
+        if (namespaceId == null) {
+            log.debug("The namespaceId was not in the namespace map, need to fetch it from the OpenShift Master.");
+            namespaceId = getProjectId(namespaceName);
+
+            // update the namespaces map if we got a hit back from the OpenShift master
+            if (namespaceId != null) {
+                namespaces.putIfAbsent(namespaceName, namespaceId);
+            }
+
+        }
+        return namespaceId;
+    }
+
+    private String getProjectId(String projectName) {
+        String projectId = null;
+        try {
+            String path = "api/v1/namespaces/" + projectName;
+            URL url = new URL(NamespaceHandler.KUBERNETES_MASTER_URL + "/" + path);
+
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            connection.setRequestProperty("Accept", "application/json");
+            connection.setRequestProperty("Authorization", "Bearer " + token);
+
+            connection.connect();
+            int responseCode = connection.getResponseCode();
+            if (responseCode == 200) {
+
+                JsonFactory jsonFactory = new JsonFactory();
+                jsonFactory.setCodec(new ObjectMapper());
+
+                InputStream inputStream = connection.getInputStream();
+                JsonParser jsonParser = jsonFactory.createParser(inputStream);
+                JsonNode jsonNode = jsonParser.readValueAsTree();
+                if (jsonNode != null) {
+                    JsonNode metadata = jsonNode.get("metadata");
+                    projectId = metadata.get("uid").asText();
+                }
+
+                jsonParser.close();
+                inputStream.close();
+
+            } else {
+                log.infof("Error getting project metadata. Response code %s", responseCode);
```

Comment:
We're not interested in the error itself with ``.getErrorStream()`` ? Or is there any possible payload?